design/src/rfcs/rfc0037_retry_classifier_customization.md

```diff
@@ -0,0 +1,240 @@
+
+```rust
+struct ExampleRetryClassifier<'a> {
+	inner: Option<&'a dyn ClassifyRetry>
+}
+
+impl<'a> ClassifyRetry for ExampleRetryClassifier<'a> {
+    fn classify_retry(
+        &self,
+        ctx: &InterceptorContext,
+        result_of_preceding_classifier: Option<RetryClassifierResult>,
+    ) -> Option<RetryClassifierResult> {
+	    // It's typical, but not required, to respect the judgement of the
+	    // preceding classifier and forward it on.
+		if let Some(result) = result_of_preceding_classifier {
+			return result;
+		}
+
+		// Do retry classification here...
+		// Assume that we found a retryable server error
+		Some(RetryClassifierResult::Error(ErrorKind::ServerError))
+	}
+}
+```
+
+When each classifier in the chain reports the result of running it, debugging the result of classification is easy:
+
+```txt
+running 'errors modeled as retryable' classifier resulted in 'continue'
+running 'retryable smithy errors' classifier resulted in 'continue'
+running 'http status code' classifier resulted in 'retry (server error)'
+```
+### The retry classifier state machine and `RetryClassifierResult`
+
+It's up to each chained classifier to respect the decision made by earlier links in the chain. When properly chained, the classifiers can be thought of as a state machine:
+
+```mermaid
+flowchart TB;
+    RetryStrategy --calls-->
+    RetryClassifier[The next retry classifier\n in the chain] --SomeError--> Retry[make a retry attempt]
+    RetryClassifier --SomeExplicit--> RetryAfter[wait for the given duration\nbefore making a retry attempt]
+    RetryClassifier --SomeDontRetry--> DontRetry[don't make a retry attempt]
+    RetryClassifier --None--> RetryClassifier
+```
+It is possible for a wrapping classifier to ignore inner classifiers, but this is not considered typical behavior. The cases where an inner classifier would be ignored MUST be clearly documented.
+
+### Setting a retry classifier with a runtime plugin
+
+Wrapping retry classifiers may be set with a runtime plugin. When setting a classifier with this method, the runtime plugin is responsible for extracting any previously-set classifier and wrapping it.
+
+```rust
+impl<'a> ExampleRetryClassifier<'a> {
+	pub fn new(inner: Option<&'a dyn ClassifyRetry>) -> Self {
+		Self { inner }
+	}
+}
+
+struct ExampleRetryClassifierRuntimePlugin;
+
+impl RuntimePlugin for ExampleRetryClassifierRuntimePlugin {
+	fn runtime_components(  
+	    &self,  
+	    current_components: &RuntimeComponentsBuilder,  
+	) -> Cow<'_, RuntimeComponentsBuilder> {
+	
+	    let rcb = RuntimeComponentsBuilder::new("ExampleRetryClassifierRuntimePlugin")
+		    .with_retry_classifier(
+			    ExampleRetryClassifier::new(current_components.retry_classifier())
+		    );
+
+	    Cow::Owned(rcb)
+	}
+}
+```
+
+By default, newer runtime plugins will override previously-set plugins. This is important to consider when deciding how your classifier will wrap other classifiers.
+
```

Comment:
I think there's a missing section on standard classifiers provided by smithy-rs, and how to reference them or disable them.